### PR TITLE
Update to KSP 1.0.1 and use new jvm modifiers resolver API

### PR DESCRIPTION
**Pull request by ZacSweers**
This allows us to fully support transient across compilation boundaries

### CODE REVIEW

**Comment by ZacSweers** [comment]
KCT needs to update https://github.com/tschuchortdev/kotlin-compile-testing/issues/222

**Review by Egorand — APPROVED** [review]

**Comment by ZacSweers** [comment]
Copying KSP support for KCT for now as they are lagging in updates

**Review by Egorand — APPROVED** [review]

**Comment by ZacSweers** [comment]
I think we need to wait for KSP 1.6.0-1.0.1 because it looks like the new effective java modifiers API is not in the 1.0.1-RC

File: kotlin/tests/extra-moshi-test-module/src/main/kotlin/com/squareup/moshi/kotlin/codegen/test/extra/AbstractClassInModuleA.kt [hunk 1]
```diff
@@ -18,9 +18,9 @@ package com.squareup.moshi.kotlin.codegen.test.extra
 import com.squareup.moshi.Json
 
 public abstract class AbstractClassInModuleA {
-  // Ignored to ensure processor sees them across module boundaries.
-  // @Transient doesn't work for this case because it's source-only and jvm modifiers aren't currently visible in KSP.
```
Comment (swankjesse) [inline kotlin/tests/extra-moshi-test-module/src/main/kotlin/com/squareup/moshi/kotlin/codegen/test/extra/AbstractClassInModuleA.kt:22]: yay

**Review by swankjesse — COMMENTED** [review]
LGTM

**Pull request merged**